hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -48,6 +49,12 @@
  *            and {@code RowData} in Flink.
  */
 public abstract class HoodieReaderContext<T> {
+  protected HoodieFileGroupReaderState<T> readerState = new HoodieFileGroupReaderState<>();
+
+  public HoodieFileGroupReaderState<T> getReaderState() {
+    return readerState;
+  }
+
```

Comment:
Can this be passed in and immutable?  It's bad idea to modify this just for tests.

Reply:
it gets modified in the reader as well